src/Uno.UI/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1348 @@
+		const int c_defaultItemSpacing = 8;
+
+		[PortStatus(Complete = true)]
+		const float c_mouseOverScale = 0.8f;
+
+		[PortStatus(Complete = true)]
+		const float c_touchOverScale = 1.0f;
+
+		[PortStatus(Complete = true)]
+		const float c_noPointerOverMagicNumber = -100;
+
+		[PortStatus(Complete = true)]
+		const float c_defaultCaptionTopMargin = 22; // 22 = 20(compensate for the -20 margin on StackPanel) + 2(magic number makes the text and star center-aligned)
+
+		[PortStatus(Complete = true)]
+		const int c_noValueSetSentinel = -1;
+
+
+		[PortStatus("Not defined by RatingControl.cpp - Jérôme's implementation", Complete = false)]
+		private SerialDisposable _subscriptions = new SerialDisposable();
+
+		[PortStatus("Not defined by RatingControl.cpp - Jérôme's implementation", Complete = false)]
+		private long m_fontFamilyChangedToken;
+
+		[PortStatus(Complete = true)]
+		public RatingControl()
+		{
+			DefaultStyleKey = GetType();
+		}
+
+		[PortStatus("Inquire about destruction", Complete = false)]
+		~RatingControl()
+		{
+			RecycleEvents();
+		}
+
+		[PortStatus(Complete = true)]
+		private float RenderingRatingFontSize()
+		{
+			// TODO MSFT #10030063 Replacing with Rating size DPs
+			return (float)(c_defaultRatingFontSizeForRendering * GetUISettings().TextScaleFactor);
+		}
+
+		[PortStatus(Complete = true)]
+		float ActualRatingFontSize()
+		{
+			return RenderingRatingFontSize() / 2;
+		}
+
+		[PortStatus(Complete = true)]
+		private double ItemSpacing()
+		{
+			// TODO MSFT #10030063: Convert to itemspacing DP
+
+			// Stars are rendered 2x size and we use expression animation to shrink them down to desired size,
+			// which will create those spacings (not system margin).
+			// Since text scale factor won't affect system margins,
+			// when stars get bigger, the spacing will become smaller.
+			// Therefore we should include TextScaleFactor when calculating item spacing
+			// in order to get correct total width and star center positions.
+			double defaultFontSize = c_defaultRatingFontSizeForRendering / 2;
+			return c_defaultItemSpacing - (GetUISettings().TextScaleFactor - 1.0) * defaultFontSize / 2;
+		}
+
+		[PortStatus(Complete = true)]
+		private void UpdateCaptionMargins()
+		{
+			// We manually set margins to caption text to make it center-aligned with the stars
+			// because star vertical center is 0.8 instead of the normal 0.5.
+			// When text scale changes we need to update top margin to make the text follow start center.
+			if (m_captionTextBlock != null)
+			{
+				double textScaleFactor = GetUISettings().TextScaleFactor;
+				Thickness margin = m_captionTextBlock.Margin;
+				margin.Top = c_defaultCaptionTopMargin - (ActualRatingFontSize() * c_verticalScaleAnimationCenterPoint);
+
+				m_captionTextBlock.Margin(margin);
+			}
+		}
+
+		[PortStatus("check var disposable = new CompositeDisposable(); isn't leaking and if (SharedHelpers::IsRS1OrHigher()). Should it even be there?", Complete = false)]
```

Comment:
TODO:
- document management of eventhandler references.
- resolve SharedHelpers::IsRS1OrHigher()